src/can/can.c

```diff
@@ -276,37 +300,124 @@ uint8_t handle_err(mob_t* mob) {
             print(ERR_MSG, "No ack");
         }
 
-        // TODO: is this the best way to handle errors?
+        // Clears CANSTMOB error bits
         CANSTMOB &= ~(0x9f);
+        //print("CANTEC: %d\n", CANTEC);
+        //print("ERRP: %d\n", CANGSTA & _BV(ERRP));
+        //print("BOFF: %d\n", CANGSTA & _BV(BOFF));
         return 1;
     }
 
     return 0;
 }
 
+// Sets baud rate for CAN (pg. 240 of data sheet)
+void set_can_rate_reg(uint8_t canbt1, uint8_t canbt2, uint8_t canbt3){
+    CANBT1 = canbt1;
+    CANBT2 = canbt2;
+    CANBT3 = canbt3;
+}
+
+// Pre-defined settings for baud rate. 100 is the default.
+void set_can_baud_rate(can_baud_rate_t baud_rate){
+    switch (baud_rate) {
+        case CAN_RATE_100:
+            set_can_rate_reg(8, 12, 55);
+            break;
+        case CAN_RATE_125:
+            set_can_rate_reg(6, 12, 55);
+            break;
+        case CAN_RATE_250:
+            set_can_rate_reg(2, 12, 55);
+            break;
+        case CAN_RATE_500:
+            set_can_rate_reg(0, 12, 54);
+            break;
+        case CAN_RATE_1000:
+            set_can_rate_reg(0, 4, 18);
+            break;
+        default:
+            break;
+    }
+}
+// Handles the circumstance where the CAN channel is not allowed to have
+// any influence on bus (i.e. entering bus off mode)
+// Reference pg. 237 for error management
+void handle_bus_off_interrupt(void){//mob_t* mob) {
+    //select_mob(mob->mob_num);
+
+    CANGIT |= _BV(BOFFIT); //setting this bit clears it
+
+    // TODO: store appropriate information to be retrieved later (e.g. last msg sent)
+    // handle interrupt by performing a software reset
+
+    if(boffit_count == 0){
+        print("Resetting CAN controller...\n");
+        CANGCON |= _BV(SWRES); // Only resets CAN controller
+        CANGCON |= _BV(ENASTB); // Enable mode (0x02)
+
+        uint16_t timeout = UINT16_MAX;
+        while (!(CANGSTA & _BV(ENFG)) && timeout > 0){ // Wait until enabled
+            print("Waiting...\n");
+            timeout--;
+        }
+        print("Node enabled...\n");
+    }
+/*
+    else if (boffit_count < 3){
+        print("Aborting message...\n");
+        CANGCON |= _BV(ABRQ); // Disable pending and terminate current communications
+    }
+*/
+/*    else {
+        print("Standby mode...\n");
+        CANGCON |= _BV(SWRES); // Only resets CAN controller
+        CANGCON |= ~(_BV(ENASTB)); // Disable node (0x02)
+
+    }
+*/
```

Comment:
Take out the conditions for disabling things, put the register writes in the comments below for reference in case we want to use them later